internal/impl/aws/input_sqs.go

```diff
@@ -412,24 +502,35 @@ func (a *awsSQSReader) readLoop(wg *sync.WaitGroup, inFlightTracker *sqsInFlight
 	}
 }
 
+type sqsMessage struct {
+	types.Message
+	handle *sqsMessageHandle
+}
+
 type sqsMessageHandle struct {
 	id, receiptHandle string
+	// The timestamp of when the message expires
+	deadline time.Time
 }
 
-func (a *awsSQSReader) deleteMessages(ctx context.Context, msgs ...sqsMessageHandle) error {
+func (a *awsSQSReader) deleteMessages(ctx context.Context, msgs ...*sqsMessageHandle) error {
+	if !a.conf.DeleteMessage {
+		return nil
+	}
+	const maxBatchSize = 10
```

Comment:
The API limit here on the amazon side is 10, so currently if you MaxNumberOfMessages > 10 then you can't delete or reset visibility for anything...